packages/jsdom-crawler/src/internals/jsdom-crawler.ts

```diff
@@ -0,0 +1,1023 @@
+            const exists = this[key as keyof this] != null;
+
+            if (!isConfigurable) { // Test if the property can be configured on the crawler
+                throw new Error(`${extension.name} tries to set property "${key}" that is not configurable on CheerioCrawler instance.`);
+            }
+
+            if (!isSameType && exists) { // Assuming that extensions will only add up configuration
+                throw new Error(
+                    `${extension.name} tries to set property of different type "${extensionType}". "CheerioCrawler.${key}: ${originalType}".`,
+                );
+            }
+
+            this.log.warning(`${extension.name} is overriding "CheerioCrawler.${key}: ${originalType}" with ${value}.`);
+
+            this[key as keyof this] = value as this[keyof this];
+        }
+    }
+
+    /**
+     * Wrapper around handlePageFunction that opens and closes pages etc.
+     */
+    protected override async _runRequestHandler(crawlingContext: CheerioCrawlingContext<JSONData>) {
+        const { request, session } = crawlingContext;
+
+        if (this.proxyConfiguration) {
+            const sessionId = session ? session.id : undefined;
+            crawlingContext.proxyInfo = await this.proxyConfiguration.newProxyInfo(sessionId);
+        }
+        if (!request.skipNavigation) {
+            await this._handleNavigation(crawlingContext);
+            tryCancel();
+
+            const { window, body, contentType, response } = await this._parseResponse(request, crawlingContext.response!);
+            tryCancel();
+
+            if (this.useSessionPool) {
+                this._throwOnBlockedRequest(session!, response.statusCode!);
+            }
+
+            if (this.persistCookiesPerSession) {
+            session!.setCookiesFromResponse(response);
+            }
+
+            request.loadedUrl = response.url;
+
+            crawlingContext.window = window!;
+            crawlingContext.contentType = contentType;
+            crawlingContext.response = response;
+            crawlingContext.enqueueLinks = async (enqueueOptions) => {
+                return cheerioCrawlerEnqueueLinks({
+                    options: enqueueOptions,
+                    window,
+                    requestQueue: await this.getRequestQueue(),
+                    originalRequestUrl: crawlingContext.request.url,
+                    finalRequestUrl: crawlingContext.request.loadedUrl,
+                });
+            };
+
+            Object.defineProperty(crawlingContext, 'json', {
+                get() {
+                    if (contentType.type !== APPLICATION_JSON_MIME_TYPE) return null;
+                    const jsonString = body!.toString(contentType.encoding);
+                    return JSON.parse(jsonString);
+                },
+            });
+
+            Object.defineProperty(crawlingContext, 'body', {
+                get() {
+                    return body;
```

Comment:
<details>
<summary>There's been a change</summary>

https://github.com/apify/apify-ts/blob/5eb05d0a24e019fae7044d00a4d11bd7ec9dd8d8/packages/cheerio-crawler/src/internals/cheerio-crawler.ts#L699-L701

This is incorrect; ifs are not memory guards and do not guarantee that something will not consume memory. Example:

```js
// node --expose-gc demo.js
const crypto = require('crypto');

const f = (() => {
        const GB = 1024 * 1024 * 1024;
        const body = crypto.randomBytes(GB * 1);

        return () => {
                if (true) {
                        return '';
                }

                // Comment the line below and it drops the buffer
                return body;
        };
})();

gc(); // Force GC to run

setTimeout(() => {
        console.log(f());
}, 10000);
```

As long as the scope where the buffer is referenced exists, the buffer is kept in the memory as well.
To free it, we'd have to directly set it to something else.
Also note that there still exists a way to retrieve the body (so this is another proof):

```js
context.contentType.type = 'application/json';
context.json; // this will error because it will try to JSON.parse(htmlContentHere)
```
</details>